src/cdk-experimental/menu/menu-item.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive, Output, Input, OnDestroy} from '@angular/core';
+import {Subject} from 'rxjs';
+import {CdkMenuPanel} from './menu-panel';
+
+/**
+ * Directive which provides behaviour for an element which when clicked either:
+ *  - executes the user defined click handler
+ *  - toggles it's checkbox state
+ *  - toggles it's radio button state (in relation to siblings)
+ *  - opens up an attached submenu
+ */
+@Directive({
+  selector: '[cdkMenuItem], [cdkMenuTriggerFor]',
+  exportAs: 'cdkMenuItem',
+  host: {
+    type: 'button',
+    '[attr.role]': 'role',
+  },
+})
+export class CdkMenuItem implements OnDestroy {
+  /** Template reference variable to the menu this trigger opens */
+  @Input('cdkMenuTriggerFor') _menuPanel: CdkMenuPanel;
+
+  /** ARIA role for the menu item. */
+  @Input() role: 'menuitem' | 'menuitemradio' | 'menuitemcheckbox' = 'menuitem';
+
+  /** Whether the checkbox or radiobutton is checked */
+  @Input() checked: boolean;
+
+  /** Emits when the attached submenu is opened */
+  @Output() opened: Subject<void> = new Subject();
+
+  /** Whether the menu item opens a menu */
+  opensMenu() {
```

Comment:
naming nit: I prefer `hasSubmenu` (or `opensSubmenu` if we want to keep the "opens" terminology)

Reply:
A sub-menu is a child of another menu. A button may trigger a menu or a sub-menu depending on where it exists in the hierarchy. 

I've used both terms when talking about it, so I could go either way, but menu is more consistent with the way the aria spec reads IMO

Reply:
Yes. The button may be in a MenuBar or Menu and `opensMenu` returns true if it opens a menu/submenu. Otherwise the button is a checkbox, radio button or just performs some action. 

`triggersMenu` is another option.